### PR TITLE
fix: Override `Event.canCoalesce()` to return `false` (M2-11083)

### PR DESCRIPTION
**Pull request by divbzero**
### 📝 Description

🔗 [Jira Ticket M2-11083](https://mindlogger.atlassian.net/browse/M2-11083)

React Native defaults `Event.canCoalesce()` to return `true` which drops earlier events of the same type when multiple are queued. This surfaced in the Android app as “Unity Error” and ”Unity Build Failed” popups.

Changes include:
- Override `Event.canCoalesce()` to return `false`.

### ✏️ Notes

Same fix as wonday/react-native-pdf#1011.

### Functionality

- [ ] The feature behaves correctly in practice and fulfills the intended business purpose
- [ ] The implementation accounts for edge cases, avoids subtle logical errors, and handles somewhat rare failure states (e.g. offline mode for mobile, 3rd party being down, etc)

### Testing

- [ ] Verify there are automated tests added that meaningfully cover critical behavior and failure cases
- [ ] Code coverage does not go down as result of this change
- [ ] Test suite passes

### Security & Data Privacy

- [ ] Verify there is no chance we would accidentally log PII to application logs
- [ ] Verify this addition does not materially affect our security attack surface, and if so it has undergone security review
- [ ] All inputs are sanitized
- [ ] New dependencies are well maintained, have significant justification for being added to the project, and are documented in the Curious [open source credit page](https://mindlogger.atlassian.net/jira/servicedesk/projects/MLA/knowledge/articles/340623543?spaceKey=MLA)

### Logging/Monitoring

- [ ] Logging is implemented for this change such that you could troubleshoot this feature in production
- [ ] The change/feature is able to be monitored in production

### Performance

- [ ] This change does not introduce n+1 queries or other performance issues within our expected scale (e.g. missing indexes on frequently queried columns, frequently updating tables that are accessed often)

### Readability

- [ ] All commented out code is removed
- [ ] Debugging code including extraneous log lines are removed
- [ ] Code is easy to understand through naming and structure; comments explain intent or non‑obvious decisions

### Change Safety

- [x] ~~Backend changes are backwards compatible with old clients, or it is well known they are not and a deployment/rollout plan is in place. This include backend changes being compatible with old mobile app versions, as well as applet versioning within Curious.~~
- [x] ~~Destructive database migrations are rolled out in stages. For example, renaming a column means adding a new column and migrating the existing data to that columns in one deployment. Then monitoring to ensure that field isn’t used, and finally removing that old column in a separate deployment.~~

### CODE REVIEW

File: src/shared/ui/BackButton.tsx [hunk 1]
```diff
@@ -1,21 +1,22 @@
 import { PropsWithChildren } from 'react';
-import { TouchableOpacity } from '@app/shared/ui/TouchableOpacity';
 
 import { useNavigation } from '@react-navigation/native';
 
+import { TouchableOpacity } from '@app/shared/ui/TouchableOpacity';
```
Comment (adeiji) [inline src/shared/ui/BackButton.tsx]: I fixed this in #1164 after you created your fix branch. Do you want to leave it as is and handle it when we merge, or should we take it out now?

**Review by adeiji — APPROVED** [review]
Looks good @divbzero !

**Comment by divbzero** [comment]
Rebased against `M2-11079-react-navigation-header-inset` to resolve conflicts in:
- `package.json`
- `yarn.lock`
- `ios/Podfile.lock`
- `patches/@azesmway+react-native-unity+1.1.1.patch`
- `src/entities/activity/ui/ActivityCard.tsx`
- `src/entities/banner/ui/Banners.tsx`
- `src/entities/unity/lib/hook/useUnityLifecycle.ts`
- `src/screens/ui/InProgressActivityScreen.tsx`
- `src/shared/lib/hooks/useStableTopInset.test.ts`
- `src/shared/lib/hooks/useStableTopInset.ts`
